nats-server/configuration/gateways/README.md

```diff
@@ -2,26 +2,30 @@
 
 ## Gateways
 
-Gateways enable connecting one or more clusters together; they allow the formation of super clusters from smaller clusters. Cluster and Gateway protocols listen in different ports. Clustering is used for adjacent servers; gateways are for joining clusters together. Typically all cluster nodes will also be gateway nodes, but this is not a requirement.
+Gateways enable connecting one or more clusters together into a full mesh; they allow the formation of super clusters from smaller clusters. Cluster and Gateway protocols listen on different ports. Clustering is used for adjacent servers; gateways are for joining clusters together. 
 
 Gateway configuration is similar to clustering:
 
 * gateways have a dedicated port where they listen for gateway requests
-* gateways gossip gateway members and remote discovered gateways
+* gateways gossip gateway nodes and remote discovered gateways
 
 Unlike clusters, gateways:
 
-* don't form a full mesh
+* have names, specifying the cluster they are in
+* don't form a full mesh between gateway nodes, but form a full mesh between cluster instead
 * are bound by uni-directional connections
+* don't gossip gateway nodes to clients
 
 Gateways exist to:
 
 * reduce the number of connections required between servers 
 * optimize the interest graph propagation
 
+If gateways are to be used in a cluster, **all** server of this cluster need to have a gateway configuration with the **same name**. Furthermore, every gateway node needs to be able to **connect to every** other gateway node and vice versa. Everything else is considered a miss configuration.
```

Comment:
> be able to **connect to every** other gateway node

That could be interpreted as a node in cluster A needs to be able to connect to all nodes in cluster B, as in creating a connection to all nodes. As you know, this is not the case. Each node needs to be able to "connect to any" other gateway node.

> miss configuration

I think "misconfiguration" instead?

Reply:
changed to any and misconfiguration